app/models/spree_amazon/address.rb

```diff
@@ -0,0 +1,64 @@
+class SpreeAmazon::Address
+  class << self
+    def find(order_reference)
+      response = mws(order_reference).fetch_order_data
+      from_response(response)
+    end
+
+    def from_response(response)
+      if response.destination["PhysicalDestination"].blank?
+        return nil
+      end
+      new attributes_from_response(response.destination["PhysicalDestination"])
+    end
+
+    private
+
+    def mws(order_reference)
+      AmazonMws.new(order_reference, test_mode)
+    end
```

Comment:
Don't really understand the point of bringing this as a private method. It only adds a level of indirection from the main method.

If `preferred_test_mode` returns true/false, then naming the method with a question mark will be better - `in_test_mode?.